pmlc/dialect/stripe/into_mlir.cc

```diff
@@ -32,18 +32,29 @@ static ScalarType DataTypeIntoMLIR(mlir::MLIRContext* ctx, DataType dtype) {  //
   return ScalarType::get(ctx, dtype);
 }
 
-static Type ShapeIntoTensorType(MLIRContext* ctx, const TensorShape& shape) {
+static Type ShapeIntoTensorType(MLIRContext* ctx, const TensorShape& shape, const stripe::Location& loc) {
   ScalarType dtype = DataTypeIntoMLIR(ctx, shape.type);
   llvm::SmallVector<TensorDim, 4> dims;
+  for (std::size_t dev_idx = 0; dev_idx < loc.devs.size(); ++dev_idx) {
+    const auto& dev = loc.devs.at(dev_idx);
+    for (std::size_t unit_idx = 0; unit_idx < dev.units.size(); ++unit_idx) {
+      dims.emplace_back(TensorDim{
+          0, 0, mlir::Identifier::get((boost::format("%s_%zu_%zu") % dev.name % dev_idx % unit_idx).str(), ctx)});
```

Comment:
2 thoughts:
1) move to use `llvm::formatv` in favor of `boost::format`
2) if staying with `boost::format`, it might be nicer to use `%1%_%2%_%3%` syntax (hmm that looks pretty ugly too).

Reply:
Done